responding to issue 2128

iamchrissmith opened the pull request:
reported in https://github.com/WP-API/WP-API/issues/2128 to check user against all visible post types

Commit:
responding to issue reported in https://github.com/WP-API/WP-API/issues/2128 to check user against all visible post types

danielbachhuber commented:
@iamchrissmith it looks like there's a merge conflict here. Can you make sure it will merge cleanly?

danielbachhuber commented on lib/endpoints/class-wp-rest-users-controller.php:
```diff
@@ -188,6 +188,13 @@ public function get_item_permissions_check( $request ) {
 		$id = (int) $request['id'];
 		$user = get_userdata( $id );
+		$types = array();
+		foreach ( get_post_types( array(), 'object' ) as $obj ) {
```
Can't we just do `$types = get_post_types( array( 'show_in_rest' => true ), 'name' )`

Commit:
Update  to just return names so it doesn't need the foreach

Commit:
resolve conflicts and update from WP-API develop branch

iamchrissmith commented:
@danielbachhuber yes, that works a lot better.  Merge conflicts also resolved (I hope)

danielbachhuber commented:
@iamchrissmith will you be adding a test for this?

Commit:
add test for get published author for pages

iamchrissmith commented:
I added a test for pages copied from the `test_get_item_published_author` and changed the name of that one to `test_get_item_published_author_post` to clarify.  I ran out of time trying to get PHPUnit setup locally to test the test, but wanted to get these up in case someone has time to test the tests

danielbachhuber commented:
Thanks for your work on this, @iamchrissmith! I've started a new PR with your branch, and fixed up the tests: #2155
